File: examples/python/submit_order.py
from decimal import Decimal

from longbridge.openapi import (
    TradeContext,
    Config,
    OAuthBuilder,
    OrderSide,
    OrderType,
    TimeInForceType,
    OutsideRTH,
)

oauth = OAuthBuilder("your-client-id").build(
    lambda url: print(f"Open this URL to authorize: {url}")
)
config = Config.from_oauth(oauth)
ctx = TradeContext(config)

resp = ctx.submit_order(
    side=OrderSide.Buy,
    symbol="700.HK",
    order_type=OrderType.MO,
    submitted_quantity=Decimal(200),
    time_in_force=TimeInForceType.Day,
    outside_rth=OutsideRTH.AnyTime,
    remark="Hello from Python SDK",
)
print(resp)

File: examples/nodejs/submit_order.js
const {
  Config,
  TradeContext,
  Decimal,
  OrderSide,
  TimeInForceType,
  OrderType,
  OAuthBuilder,
} = require("longport");

async function main() {
  const oauth = await OAuthBuilder.build("your-client-id", (url) => {
    console.log("Open this URL to authorize: " + url);
  });
  let config = Config.fromOAuth(oauth);
  let ctx = await TradeContext.new(config);
  let resp = await ctx.submitOrder({
    symbol: "700.HK",
    orderType: OrderType.LO,
    side: OrderSide.Buy,
    timeInForce: TimeInForceType.Day,
    submittedPrice: new Decimal(50),
    submittedQuantity: new Decimal(200),
  });
  console.log(resp.toString());
}

Promise.all([main()]).catch((err) => console.error(err));
